Ability to compute the maximum absolute value in ExtremeValue PPs

zachmprince opened the pull request:
## Reason

Closes #29745

## Design

Simple enum addition in the base class with another logic tree for storing the absolute value of the proxy value.

## Impact

New capability

<!--
If this PR implements a bug fix, please create an issue for the bug and reference the issue.
-->

GiudGiud committed:
Add support for maximum absolute value in ExtremeValue PPs
closes #29745

zachmprince committed:
Adding test for maximum absolute value extreme PP #29745

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2716223/), step Docs: sync website on ac0c29c wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/29976/site)

This comment will be updated on new commits.

zachmprince committed:
Fixing parallelness #29745

GiudGiud commented on framework/src/postprocessors/ExtremeValueBase.C:
```diff
@@ -96,8 +86,8 @@ ExtremeValueBase<T>::threadJoin(const UserObject & y)
-       pps._proxy_value > _proxy_value) ||
-      (_type == ExtremeType::MIN && pps._proxy_value < _proxy_value))
+       pps._proxy_value.first > _proxy_value.first) ||
+      (_type == ExtremeType::MIN && pps._proxy_value.first < _proxy_value.first))
```
I think this will be less consistent in parallel in the cases where you have equality in the proxY_value ?

zachmprince replied:
Oh! I didn't think about that. I'll revert this change.

GiudGiud replied:
sounds good. I added the sweep recipes to make sure test suite passes in either configuration

zachmprince committed:
Reverting change to login in threadJoin #29745

lindsayad committed:
Fix doxygen, parameter names for UserObject proxy methods

The parameter names were exactly backwards before

lindsayad commented on framework/src/postprocessors/ExtremeValueBase.C:
```diff
@@ -71,20 +73,10 @@ template <class T>
 void
 ExtremeValueBase<T>::finalize()
 {
-  if (_type == ExtremeType::MAX)
-  {
-    if (_use_proxy)
-      this->gatherProxyValueMax(_proxy_value.first, _proxy_value.second);
-    else
-      this->gatherMax(_proxy_value.second);
-  }
```
if we're going to do this simplification then we can just remove `_use_proxy`. I don't think there's much of a performance penalty here so I'm fine with this

lindsayad commented on test/tests/postprocessors/element_extreme_functor_value/tests:
```diff
@@ -19,4 +19,10 @@
+  [abs_extreme]
+    type = RunApp
+    input = extreme_abs_value.i
+    requirement = 'The system shall compute the extreme absolute value of a functor over the domain.'
```
Mention something about proxy capability as well

lindsayad reviewed (changes requested):
@zachmprince please review the commit I pushed

zachmprince committed:
Addressing review

- Updating doc-string of 'value_type' parameter
- Removing _use_proxy
- Clarifying requirement in test

moosebuild commented:
Job [Test HPC](https://civet.inl.gov/job/2728922/) on 009d358 : invalidated by @zachmprince

HPC issues have been resolved
